service/src/main/java/com/commercetools/pspadapter/payone/mapping/SofortBankTransferRequestFactory.java

```diff
@@ -50,8 +47,8 @@ public BankTransferAuthorizationRequest createAuthorizationRequest(final Payment
 
         if (StringUtils.isNotEmpty(ctPayment.getCustom().getFieldAsString(CustomFieldKeys.BIC_FIELD))) {
             final String plainBic;
-            if (!serviceConfig.getSecureKey().isEmpty()) {
-                plainBic = BlowfishUtil.decryptHexToString(serviceConfig.getSecureKey(), ctPayment.getCustom().getFieldAsString(CustomFieldKeys.BIC_FIELD));
+            if (!secureKey.isEmpty()) {
+                plainBic = BlowfishUtil.decryptHexToString(secureKey, ctPayment.getCustom().getFieldAsString(CustomFieldKeys.BIC_FIELD));
```

Comment:
maybe better if those two if condition blocks setting the iban field and bic fields of the request are there in their separate methods